scripts/gen_architecture_overview.py

```
@@ -0,0 +1,175 @@
+out = []
+
+
+def esc(s):
+    return s.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;")
+
+
+def fs(t):
+    return 11.5 if len(t) > 13 else (12.5 if len(t) > 10 else 14)
+
+
+def box(cx, y, label, fill, edge, link=None, cls="node", bold=False, w=BOX_W, h=BOX_H):
```

Ruff (0.15.21)

[warning] Boolean default positional argument in function definition

(FBT002)